src/ctia/entity/relationship.clj

```diff
@@ -90,98 +91,100 @@
-              source-short-id (-> link-req
-                                  source-type-kw
-                                  long-id->id
-                                  :short-id)
-              source-store-kw (case source-type-kw
-                                :casebook_id :casebook
-                                :investigation_id :investigation)
-              source (when source-short-id
-                       (read-store source-store-kw
-                                   read-record
-                                   source-short-id
-                                   identity-map
-                                   {}))
-              target-ref (short-id->long-id id services)]
-          (cond
-            (or (not incident)
-                (not target-ref))
-            (not-found {:error "Invalid Incident id"})
-            (not source)
-            (bad-request {:error (str "Invalid "
-                                      (case source-type-kw
-                                        :casebook_id "Casebook"
-                                        :investigation_id "Investigation")
-                                      " id")})
-            :else
-            (let [{source-id source-type-kw
-                   :keys [tlp]
-                   :or {tlp "amber"}} link-req
-                  new-relationship
-                  {:source_ref source-id
-                   :target_ref target-ref
-                   :relationship_type "related-to"
-                   :tlp tlp}
-                  stored-relationship
-                  (-> (flows/create-flow
-                       :services services
-                       :entity-type :relationship
-                       :realize-fn rs/realize-relationship
-                       :store-fn #(write-store :relationship
-                                               create-record
-                                               %
-                                               identity-map
-                                               {})
-                       :long-id-fn #(with-long-id % services)
-                       :entity-type :relationship
-                       :identity identity
-                       :entities [new-relationship]
-                       :spec :new-relationship/map)
-                      first
-                      un-store)]
-              (created stored-relationship))))))
+  (let [;; :read-<source-type> is checked while processing the body
+        capabilities #{:read-incident
+                       :read-relationship
+                       :create-relationship}]
+    (POST "/:id/link" []
+          :return rs/Relationship
+          :body [link-req IncidentLinkRequest
+                 {:description "an Incident Link request"}]
+          :summary "Link an Incident to a Casebook or Investigation"
+          :path-params [id :- s/Str]
+          :description (common/capabilities->description capabilities)
```

Comment:
You could modify this description to precise the additional required capabilities.
``` clojure
(str (common/capabilities->description capabilities)
       "read-casebook or read-investigation are also respectively required for linked casebook or investigation.")
```
(or anything else less franglish :-) )

Reply:
Nice catch! Generated this docstring by moving some data around.

<img width="622" alt="Screenshot 2020-11-17 at 8 02 20 PM" src="https://user-images.githubusercontent.com/287396/99469018-e51c0880-290f-11eb-91a2-7a0091ea8bd1.png">